ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/fillers/CoreProblemFiller.java

```diff
@@ -56,10 +69,39 @@ public void fill(RaoData raoData, LinearProblem linearProblem) {
 
     @Override
     public void update(RaoData raoData, LinearProblem linearProblem) {
+        // chose range actions to use
+        computeAvailableRangeActions(raoData);
         // update reference flow and sensitivities of flow constraints
         updateFlowConstraints(raoData, linearProblem);
     }
 
+    /**
+     * If a TSO has a maximum number of usable ranges actions, this functions filters out the range actions with
+     * the least impact on the most limiting element
+     */
+    private void computeAvailableRangeActions(RaoData raoData) {
+        availableRangeActions = raoData.getAvailableRangeActions();
+        if (!Objects.isNull(maxPstPerTso) && !maxPstPerTso.isEmpty()) {
+            RaoParameters.ObjectiveFunction objFunction = raoData.getRaoParameters().getObjectiveFunction();
+            BranchCnec mostLimitingElement = RaoUtil.getMostLimitingElement(raoData.getCnecs(), raoData.getWorkingVariantId(), objFunction.getUnit(), objFunction.relativePositiveMargins());
+            maxPstPerTso.forEach((String tso, Integer maxPst) -> {
+                Set<RangeAction> rangeActionsForTso = availableRangeActions.stream().filter(rangeAction -> rangeAction.getOperator().equals(tso)).collect(Collectors.toSet());
```

Comment:
add a filter`(rangeAction -> rangeAction instanceof PstRange)`
might be usefull one day

Reply:
yes !